graphql/fragments.graphql

```diff
@@ -6,6 +6,8 @@ fragment UserFragment on User {
   bio
   userRole
   profileImage
+  postsWrittenCount
+  thanksReceivedCount
```

Comment:
I think this should go on another fragment, maybe `UserWithStatsFragment` like:

```
fragment UserWithStatsFragment {
  ...UserFragment
  postsWrittenCount
  thanksReceivedCount
}
```

and use that on the post page so we don't have to spend the cycles/queries getting these numbers in places where we don't use them (e.g. current user, or when fetching post cards)

Reply:
Yep I was a bit unsure of this and had it in my notes to ask your opinion about haha, great suggestion. The only trouble here is that we're using the `AuthorWithLanguagesFragment` in the `PostAuthorCard` that you're referencing. I'm actually a bit unsure what to do there... take a look at my new changes in [this commit](https://github.com/Journaly/journaly/pull/336/commits/f47f94553a49f57426ec9b15364e38385bf07ba8) and let me know you're thoughts

Reply:
Hmm, I see, we do have a certain proliferation of fragments. I'm ok with this approach, we might consider taking an approach of each query having its own fragment, so `PostByIdFragment` could combine "AuthorWithLanguagesFragment" with the two stats. Or maybe an "AuthorCardFragment". I think the normal graphql approach would be to have a "AuthorFragment" and anything extra you need just goes in the query, but we need extra fragments because that gives us a named type. Something to think about, but as long as we're not pulling `postsWrittenCount` and `thanksReceivedCount` on like the feed and other places we're using `PostCard` it's good with me.